src/core/database/common.repository.ts

```diff
@@ -13,7 +14,31 @@ export class CommonRepository {
     return await this.db.checkDeletePermission(id, session);
   }
 
-  async deleteNode(objectOrId: { id: ID } | ID) {
-    await this.db.deleteNode(objectOrId);
+  async deleteNode(objectOrId: { id: ID } | ID, changeset?: ID) {
+    if (!changeset) {
+      await this.db.deleteNode(objectOrId);
+    } else {
```

Comment:
Would you mind returning above and removing the else statement? It's a pet peeve of mine 😬